core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java

```diff
@@ -239,7 +239,7 @@ public Map<K, V> getAll(Set<?> keys) {
       }
       for (Object k : keys) {
          Objects.requireNonNull(k, NULL_KEYS_NOT_SUPPORTED);
-         InternalCacheEntry<K, V> entry = getDataContainer().get(k);
+         InternalCacheEntry<K, V> entry = getDataContainer().peek(k);
```

Comment:
I don't think you can use `peek` here as it doesn't update the access time.

Reply:
Actually I think `getAsync` isn't working properly, so we will need to copy over the logic from `AbstractInternalDataContainer#get` for the simple cache like you originally proposed.